web_poet/exceptions/http.py

```diff
@@ -6,14 +6,59 @@
 operations.
 """
 
+from web_poet.page_inputs import HttpResponse, HttpRequest
 
-class HttpRequestError(IOError):
-    """Indicates that an exception has occurred when the HTTP Request was being
-    handled.
 
-    For responses that are successful but don't have a ``200`` **status code**,
-    this exception shouldn't be raised at all. Instead, the :class:`~.HttpResponse`
-    should simply reflect the response contents as is.
+class HttpError(IOError):
+    """Indicates that an exception has occurred when handling an HTTP operation.
+
+    This is used as a **base class** for more specific errors and could be vague
+    since it could denote problems either in the HTTP Request or Response.
+
+    For more specific errors, it would be better to use :class:`.HttpRequestError`
+    and :class:`.HttpResponseError`.
+
+    :param request: The :class:`~.HttpRequest` instance that was used.
+    :type request: HttpRequest
+    :param response: The :class:`~.HttpResponse` instance that was received, if
+        any. Note that this wouldn't exist if the problem ocurred when executing
+        the :class:`~.HttpRequest`.
+    :type response: HttpResponse
```

Comment:
I wonder if response should be an attribute of HttpResponseError, not an attribute of a base class. What's a use case for having it here, and in HttpRequestError?

Reply:
Thanks for raising this! I've updated this on https://github.com/scrapinghub/web-poet/pull/38/commits/a209b4d83932261026937c9f27122d987d97f550